hoot-services/src/main/java/hoot/services/controllers/clipping/UpdateParentCommand.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2016, 2017, 2018, 2019 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+package hoot.services.controllers.clipping;
+
+import static hoot.services.models.db.QFolderMapMappings.folderMapMappings;
+import static hoot.services.utils.DbUtils.createQuery;
+
+import java.time.LocalDateTime;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import hoot.services.command.CommandResult;
+import hoot.services.command.InternalCommand;
+import hoot.services.controllers.osm.map.FolderResource;
+import hoot.services.controllers.osm.map.MapResource;
+import hoot.services.models.osm.Map;
+import static hoot.services.models.db.QJobStatus.jobStatus;
+import hoot.services.models.db.Users;
+import hoot.services.utils.DbUtils;
+
+
+class UpdateParentCommand implements InternalCommand {
+    private static final Logger logger = LoggerFactory.getLogger(UpdateParentCommand.class);
+
+    private final String mapName;
+    private final String jobId;
+    private final Users user;
+    private final Long folderId;
+    private final Class<?> caller;
+
+    UpdateParentCommand( String jobId, ClipDatasetParams params, Users user, Class<?> caller) {
+        this.mapName = params.getOutputName();
+        this.jobId = jobId;
+        this.user = user;
+        this.folderId = params.getFolderId();
+        this.caller = caller;
+    }
+
+    @Override
+    public CommandResult execute() {
+        CommandResult commandResult = new CommandResult();
+        commandResult.setJobId(jobId);
+        commandResult.setCommand("[Update Parent Directory] of map with name = " + mapName);
+        commandResult.setStart(LocalDateTime.now());
+        commandResult.setCaller(caller.getName());
+
+        updateParentDirectory();
+
+        commandResult.setFinish(LocalDateTime.now());
+        commandResult.setExitCode(CommandResult.SUCCESS);
+
+        return commandResult;
+    }
+
+    private void updateParentDirectory() {
+        try {
+            DbUtils.updateFoldersMapping();
+
+            Long mapId = createQuery()
+                    .select(jobStatus.resourceId)
+                    .from(jobStatus)
+                    .where(jobStatus.jobId.eq(jobId)).fetchOne();
+
+            // These functions ensure the map + folder are either owned by the user -or- public.
+            Map m = MapResource.getMapForUser(user, mapId.toString(), false, true);
+            FolderResource.getFolderForUser(user, folderId);
+
+            // Delete any existing to avoid duplicate entries
+            createQuery().delete(folderMapMappings).where(folderMapMappings.mapId.eq(m.getId())).execute();
```

Comment:
I wonder why map id wasn't made the primary key in the folder map mappings table?  Is this done elsewhere?

Reply:
I see it in FolderResource.  We should consolidate this with the queries there in a FolderManager.

Reply:
So it's being used here : https://github.com/ngageoint/hootenanny/blame/develop/hoot-services/src/main/java/hoot/services/controllers/osm/map/MapResource.java#L651

The idea of it not being a primary key doesnt make sense to me either since mapid seems to be generated from a sequence based on what iv'e seen in the job_status table.

Will consolidate to folder manager. I think it will be helpful since I already see how it can be used in other classes